typing/typecore.ml

```diff
@@ -380,7 +407,12 @@ let mode_argument ~funct ~index ~position ~partial_app alloc_mode =
                 Id_prim _), 1, Tail ->
      (* The second argument to (&&) and (||) is in
         tail position if the call is *)
-     mode_return (Value_mode.local_to_regional vmode)
+      (* CR zqian: this is suspicious,
+         because vmode is not the mode of the region;
+         ideally the `position` should contain the mode
+         of the region so we can pass it to mode_return;
+         but this might be fine because of mode-crossing? *)
```

Comment:
I agree this is dubious. I wonder if we can treat `%sequor` and `%sequand` differently elsewhere rather than here. Maybe handling them where we handle `%ignore` in `type_application` would be easier.